test/extension/handlers/topologymutation/handler.go

```diff
@@ -19,63 +19,225 @@ package topologymutation
 
 import (
 	"context"
+	"fmt"
 	"strconv"
+	"strings"
 
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
 	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/serializer"
+	intstrutil "k8s.io/apimachinery/pkg/util/intstr"
 	ctrl "sigs.k8s.io/controller-runtime"
 
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
 	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
 	patchvariables "sigs.k8s.io/cluster-api/internal/controllers/topology/cluster/patches/variables"
 	infrav1 "sigs.k8s.io/cluster-api/test/infrastructure/docker/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/version"
 )
 
 // NewHandler returns a new topology mutation Handler.
 func NewHandler(scheme *runtime.Scheme) *Handler {
 	return &Handler{
+		// Add the types being handled to the decoder
 		decoder: serializer.NewCodecFactory(scheme).UniversalDecoder(
 			infrav1.GroupVersion,
+			controlplanev1.GroupVersion,
+			bootstrapv1.GroupVersion,
 		),
 	}
 }
 
+var cgroupfs = "cgroupfs"
+
 // Handler is a topology mutation handler.
 type Handler struct {
 	decoder runtime.Decoder
 }
 
+var logger logr.Logger
+
 // GeneratePatches returns a function that generates patches for the given request.
```

Comment:
```suggestion
// GeneratePatches generates patches for the given request.
```
Sorry my bad. The comment was from a previous version of the func where it actually returned a func. (same for ValidateTopology)